Use MongoWriteBatch, WriteConcern & Container namespace

prolic opened the pull request:
see: https://github.com/prooph/event-store-mongodb-adapter/issues/12, https://github.com/prooph/event-store-mongodb-adapter/issues/10 and https://github.com/prooph/event-store/issues/69

prolic committed:
Rename *Service* namespace to *Container*

prolic committed:
Fix test suite in case mongod is not running

prolic committed:
Add write concern to constructor

prolic committed:
Use write concern in adapter

prolic committed:
Make use of MongoInsertBatch class

see: https://github.com/prooph/event-store-mongodb-adapter/issues/10

prolic committed:
Make use of MongoInsertBatch, MongoUpdateBatch and MongoDeleteBatch

codeliner commented:
:boom: 

``` batch
MongoProtocolException: Current primary does not have a Write API support
```

Bigger problem?

prolic commented:
Seems like travis has an old driver or old db version. I'll check that.
Am 21.08.2015 19:40 schrieb "Alexander Miertsch" notifications@github.com:

> [image: :boom:]
> 
> MongoProtocolException: Current primary does not have a Write API support
> 
> Bigger problem?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/prooph/event-store-mongodb-adapter/pull/17#issuecomment-133527159
> .

codeliner commented:
ok, we should add an info to the README which versions of the driver and db are required to use the adapter.

codeliner commented:
Btw: Do you know if it is possible to `require` the `mongo.so` extension in composer.json with the minimal version the adapter needs?

prolic commented:
I will test it and submit to this PR.

prolic committed:
test travis mongo versions

prolic committed:
test travis mongo versions

prolic committed:
test travis mongo versions

prolic commented:
Travis has mongo db version v2.4.12 (way to old!). I'll update the travis script.

prolic committed:
Install mongo db 3.0.5 on travis

prolic committed:
Travis mongo download without https

prolic committed:
update README, composer.json, fix travis

prolic committed:
fix php cs

prolic commented:
finally green! ;)

prolic committed:
fix README typo

codeliner commented on src/Container/MongoDbEventStoreAdapterFactory.php:
```diff
@@ -44,6 +44,6 @@ public function __invoke(ContainerInterface $container)
         $streamCollectionName = isset($adapterOptions['collection_name']) ? $adapterOptions['collection_name'] : null;
 
-        return new MongoDbEventStoreAdapter($mongoClient, $dbName, $streamCollectionName);
+        return new MongoDbEventStoreAdapter($mongoClient, $dbName, [], $streamCollectionName);
```
A config option for the write concern (`write_concern`) would be nice

codeliner commented on src/MongoDbEventStoreAdapter.php:
```diff
@@ -72,6 +106,8 @@ public function __construct(\MongoClient $mongoClient, $dbName, $streamCollectio
             $this->streamCollectionName = $streamCollectionName;
         }
+
+        $this->writeConcern = array_merge($this->writeConcern, $writeConcern);
```
hhm, can this cause trouble if one is not aware that writeConcern is merged instead of replaced? Should we document it?

codeliner commented:
great job :+1: It is really cool to have such a high quality adapter for Mongo DB available.
Just two minor questions/notes.

prolic committed:
Inject write concern in factory

prolic committed:
Allow custom transaction timeout

additionally the write concern option is no longer merged, but overwritten
updated readme

prolic committed:
Add transaction timeout to factory

prolic commented:
I added _write_concern_ and _transaction_timeout_ as options to the factory. The write concern is also no longer merged. Default values are documented in the readme.

prolic committed:
Fix typo in readme

prolic commented:
Question: What if a transaction has already begun and the client code asks to open another one? Should we throw an exception?

prolic committed:
Add test for transaction timeout rollback

codeliner commented:
Exception sounds good

prolic committed:
Second transaction start throws exception

prolic commented:
all done.

codeliner commented:
I guess this will be my new default adapter for the event store :smile:

codeliner commented:
as soon as 5.0 is out
